tools/codegen/api/python.py

```diff
@@ -817,27 +832,15 @@ def arg_parser_output_expr(
     #
     # TODO: this seems too complicated - maybe we can simplify after full
     # c10 dispatch migration?
-    typename = la.cpp_type_str \
-        if a.name != 'out' and la is not None else a.cpp_type_str
-
-    if a.default_init is not None:
-        # Note: only introduced in tensor_options_args
-        if typename not in UNPACK_WITH_DEFAULT_METHODS:
-            raise RuntimeError(
-                f'type \'{typename}\' is not supported in default_init')
-        unpack_with_default = UNPACK_WITH_DEFAULT_METHODS[typename]
-        expr = f'_r.{unpack_with_default}({arg_index}, {a.default_init})'
-    elif a.size is not None:
-        if typename not in UNPACK_WITH_SIZE_METHODS:
-            raise RuntimeError(
-                f'type \'{typename}\' with definite size ({a.size}) is not supported')
-        unpack_with_size = UNPACK_WITH_SIZE_METHODS[typename].format(a.size)
-        expr = f'_r.{unpack_with_size}({arg_index})'
+    if a.name != 'out' and la is not None and isinstance(la.argument, Argument):
+        arg_type = la.argument.type
```

Comment:
This still seems a bit more complicated than it should be

Reply:
> This still seems a bit more complicated than it should be

Yes, I figured we don't need use la.argument.type anymore as now we can directly derive it from 'local.use_c10_dispatcher().dispatcher_uses_new_style()' :)

Deleted it and the long comment above!